File: iosApp/iosApp/core/util/ext/MonthExt.swift
//
//  MonthExt.swift
//  iosApp
//
//  Created by Mikhail Maevskii on 19/12/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import SwiftUI
import Shared

extension Month {
    func getName() -> String {
        return switch self.number {
        case 1: NSLocalizedString("january", comment: "")
        case 2: NSLocalizedString("february", comment: "")
        case 3: NSLocalizedString("march", comment: "")
        case 4: NSLocalizedString("april", comment: "")
        case 5: NSLocalizedString("may", comment: "")
        case 6: NSLocalizedString("june", comment: "")
        case 7: NSLocalizedString("july", comment: "")
        case 8: NSLocalizedString("august", comment: "")
        case 9: NSLocalizedString("september", comment: "")
        case 10: NSLocalizedString("october", comment: "")
        case 11: NSLocalizedString("november", comment: "")
        case 12: NSLocalizedString("december", comment: "")
        default: NSLocalizedString("error_getting_current_month", comment: "")
        }
    }
}

File: shared/src/commonMain/kotlin/dem/dev/timeflame/util/datetime/MonthExt.kt
package dem.dev.timeflame.util.datetime

import kotlinx.datetime.Month

fun Month.getDaysAmount(year: Int): Int {
    return when(this) {
        Month.JANUARY -> 31
        Month.FEBRUARY -> if (isLeapYear(year)) 29 else 28
        Month.MARCH -> 31
        Month.APRIL -> 30
        Month.MAY -> 31
        Month.JUNE -> 30
        Month.JULY -> 31
        Month.AUGUST -> 31
        Month.SEPTEMBER -> 30
        Month.OCTOBER -> 31
        Month.NOVEMBER -> 30
        Month.DECEMBER -> 31
        else -> 0
    }
}

fun isLeapYear(year: Int): Boolean {
    return (year % 100 != 0 && year % 4 == 0) || (year % 100 == 0 && year % 400 == 0)
}